movieApp.js

```diff
@@ -268,3 +268,94 @@ function searchMovies(keyword) {
 }
 
 showMovies(moviesDatabase);
+
+// Set the timer
+document.addEventListener('DOMContentLoaded', () => {
+  const startButton = document.getElementById('start-btn');
+  const timeInput = document.getElementById('time-input');
+  const countdownElement = document.getElementById('countdown');
+  const timeUpPopup = document.getElementById('time-up-popup');
+  const continueButton = document.getElementById('continue-btn');
+  const randomButton = document.getElementById('random-btn');
+  const closeTimerBlock = document.getElementById('close-timer-block');
+  const timerBlock = document.getElementById('timer-block');
+  const timerNavbarIcon = document.querySelector('.toggle-timer-block');
+
+  const timerAlarm = new Audio('tazflix-alarm.wav');
+  let timer;
+
+  startButton.addEventListener('click', () => {
+      const timeValue = parseInt(timeInput.value);
+      if (isNaN(timeValue) || timeValue <= 0) {
+          alert('Please enter a valid time in minutes.');
+          return;
+      }
+
+      clearInterval(timer);
+      let timeLeft = timeValue * 60;
+
+      timer = setInterval(() => {
+          if (timeLeft <= 0) {
+              clearInterval(timer);
+              showPopup();
+              countdownElement.textContent = '';
+              timerAlarm.play();
+          } else {
+              timeLeft--;
+              const minutes = Math.floor(timeLeft / 60);
+              const seconds = timeLeft % 60;
+              countdownElement.textContent = `${minutes}m ${seconds}s remaining`;
+          }
+      }, 1000);
+  });
+
+  continueButton.addEventListener('click', () => {
+      hidePopup();
+  });
+
+  randomButton.addEventListener('click', () => {
+      hidePopup();
+      chooseRandomMovie();
+  });
+
+  closeTimerBlock.addEventListener('click', () => {
+      timerBlock.classList.add('hidden');
+  });
+
+  timerNavbarIcon.addEventListener('click', () => {
+      if (timerBlock.classList.contains('hidden')) {
```

Comment:
You can simplify this by using [toggle](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle)